File: backend/internal/models/system.go
package models

import "time"

// System representa un sistema monitoreado
type System struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Type        string    `json:"type"`        // "web", "api", "google-script"
	Environment string    `json:"environment"` // "prod", "preprod"
	Status      string    `json:"status"`      // "online", "offline", "degraded", "unknown"
	LastCheck   time.Time `json:"last_check"`
	Checks      []Check   `json:"checks"`
}

// Check representa una verificación individual
type Check struct {
	ID           string                 `json:"id"`
	Type         string                 `json:"type"`    // "http", "database", "login", "custom"
	Name         string                 `json:"name"`    // Nombre descriptivo del check
	Status       string                 `json:"status"`  // "ok", "warning", "error"
	Message      string                 `json:"message"` // Descripción del estado
	LastCheck    time.Time              `json:"last_check"`
	ResponseTime int64                  `json:"response_time_ms"` // Tiempo de respuesta en ms
	Metadata     map[string]interface{} `json:"metadata,omitempty"`
}

File: frontend/src/types/system.ts
// Tipos basados en los modelos del backend Go

export type SystemStatus = 'online' | 'warning' | 'error' | 'unknown';
export type Environment = 'prod' | 'preprod' | 'shared';
export type CheckType = 'http' | 'database' | 'rdap' | 'google-sheets';

export interface Check {
  id: string;
  type: CheckType;
  name: string;
  status: SystemStatus;
  message: string;
  last_check: string; // ISO date string
  response_time_ms: number;
  metadata?: Record<string, any>;
}

export interface System {
  id: string;
  name: string;
  type: string;
  environment: Environment;
  status: SystemStatus;
  last_check: string; // ISO date string
  checks: Check[];
  // Campos para rastrear origen de datos (frontend only)
  source?: 'cache' | 'sse';
  localUpdatedAt?: Date;
}

export interface SystemsResponse {
  systems: System[];
  cached: boolean;
  count: number;
}

export interface RefreshResponse {
  message: string;
  status: string;
  system_id?: string;
}

// SSE Event types
export interface SSEConnectedEvent {
  client_id: string;
}

export interface SSECheckCompleteEvent {
  message: string;
}

// Tipos auxiliares para UI
export interface SystemStats {
  total: number;
  online: number;
  warning: number;
  error: number;
  onlinePercentage: number;
  warningPercentage: number;
  errorPercentage: number;
}

export interface CheckMetadata {
  [key: string]: string | number | boolean | string[] | undefined;
}
